.eslintrc.js

```diff
@@ -1,39 +1,54 @@
+const environmentRules = require('@metamask/eslint-config/src/environment.json');
+
 module.exports = {
   root: true,
 
-  extends: [
-    '@metamask/eslint-config',
-    '@metamask/eslint-config/config/mocha',
-    '@metamask/eslint-config/config/nodejs',
-  ],
-
-  parser: 'babel-eslint',
-
-  parserOptions: {
-    ecmaVersion: 2017,
-  },
-
-  plugins: ['json', 'import'],
-
-  globals: {
-    document: 'readonly',
-    window: 'readonly',
-  },
+  extends: ['@metamask/eslint-config'],
 
   overrides: [
     {
-      files: ['.eslintrc.js'],
+      files: ['*.ts'],
+      extends: ['@metamask/eslint-config-typescript'],
+    },
+
+    {
+      files: ['*.js'],
       parserOptions: {
         sourceType: 'script',
       },
+      extends: ['@metamask/eslint-config-nodejs'],
+    },
+
+    {
+      files: ['*.test.ts'],
+      extends: ['@metamask/eslint-config-mocha'],
     },
   ],
 
+  rules: {
+    'import/no-nodejs-modules': 'off',
+    'no-restricted-globals': [
```

Comment:
Hmm. It's too bad that there's not another way to do this. That said, nice job for reusing the list already defined in `@metamask/eslint-config` — we should use that technique more often. 

@Gudahtt Do you have any suggestions for how we could make this better?

Reply:
We could use this package, if we want to require this only be used in-browser: https://github.com/MetaMask/eslint-config/tree/main/packages/browser

If we're anticipating this will be used in a Node.js context, with polyfills for each referenced browser global, than this approach is probably best. The alternative would be to eslint-ignore each individual violation (though maybe we could do that in a nice way by bringing them all to the top of the module and reassigning them to variables).

Reply:
In some lines, we use the `Buffer` Node.js built-in without using any particular import to handle that, and indeed after using `@metamask/eslint-config-browser` those `Buffer.from(...)` ar marked as error.

Adding `import { Buffer } from 'buffer';` should do the trick but I'm unsure how it worked before now. Is that just a TS requirement in your opinion?

Reply:
Oh sorry, I misunderstood. So that global is available in Node.js and injected by the extension build system, that's why it had worked before. It would be ideal to remove that from this package, but allowing that global as an exception or importing it directly should work instead for now (either should be equivalent)

Reply:
This is a good example of why this browser ESLint package is useful; it can highlight undocumented requirements like this. Now we know that this package assumes there is a `Buffer` polyfill present. We could document that requirement in the README so third-parties know about that requirement.